controllers/database_reconciler_helpers.go

```diff
@@ -145,59 +146,18 @@ func (r *DatabaseReconciler) handleDelete(ctx context.Context, database *ndbv1al
 	return requeueWithTimeout(15)
 }
 
-// When a database is provisioned, an id is assigned to the database CR.
-// In case if someone deletes the database externally/aborts the operation through NDB (and not through the operator), the operator should
-// create a new database. To do this, we fetch the database by the Id we have in the datbase CR's Status.Id.
-// If the database response's status is empty, we set our CR's status to be empty so that it is provisioned again.
-func (r *DatabaseReconciler) handleExternalDelete(ctx context.Context, database *ndbv1alpha1.Database, ndbClient *ndb_client.NDBClient) (err error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Entered database_reconciler_helpers.handleExternalDelete")
-	if database.Status.Id != "" {
-		// database was provisioned earlier => sync status
-		var databaseResponse ndb_api.DatabaseResponse
-		allDatabases, err := ndb_api.GetAllDatabases(ctx, ndbClient)
-		if err != nil {
-			errStatement := "Error fetching all databases from NDB"
-			log.Error(err, errStatement)
-			// CHECK
-			r.recorder.Eventf(database, "Warning", EVENT_RESOURCE_LOOKUP_ERROR, "Error: %s", errStatement, err)
-
-			return err
-		} else {
-			for _, db := range allDatabases {
-				if db.Id == database.Status.Id {
-					databaseResponse = db
-					break
-				}
-			}
-		}
-		// Update the CR status if the database response is empty so that it triggers a provision operation
-		if databaseResponse.Status == common.DATABASE_CR_STATUS_EMPTY {
-			log.Info("The database might have been deleted externally, setting an empty status so it can be re-provisioned.")
-			r.recorder.Event(database, "Normal", EVENT_EXTERNAL_DELETE, "The database has been deleted externally (on NDB). Reprovisioning database on NDB.")
-			database.Status.Status = common.DATABASE_CR_STATUS_EMPTY
-			err = r.Status().Update(ctx, database)
-			if err != nil {
-				errStatement := "Failed to update status of database custom resource"
-				log.Error(err, errStatement)
-				r.recorder.Eventf(database, "Warning", EVENT_CR_STATUS_UPDATE_FAILED, "Error: %s. %s.", err.Error())
-				return err
-			}
-		}
-
-	}
-	return nil
-}
-
 // The handleSync function synchronizes the database CR's with the database instance in NDB
```

Comment:
Can you update this comment, as we are now syncing with the NDB CR rather than the NDB instance directly